main.go

```diff
@@ -268,8 +299,17 @@ func metricsServer(registry prometheus.Gatherer, port int) {
 	mux.Handle("/debug/pprof/symbol", http.HandlerFunc(pprof.Symbol))
 	mux.Handle("/debug/pprof/trace", http.HandlerFunc(pprof.Trace))
 
+	promhandler := promhttp.HandlerFor(registry, promhttp.HandlerOpts{})
 	// Add metricsPath
-	mux.Handle(metricsPath, promhttp.HandlerFor(registry, promhttp.HandlerOpts{}))
+	mux.Handle(metricsPath, http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		ok := AuthRequest(auth, w, req)
+		if !ok {
```

Comment:
Maybe a 401 response status code and some description in the response body is more user friendly and directly.

Reply:
By just `return`, golang http server will return response status code with 200 and nothing response body which is quite confused.

Reply:
Added.

Reply:
lgtm